crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -69,20 +70,36 @@ pub fn derive_bundle(input: TokenStream) -> TokenStream {
     let mut field_component_ids = Vec::new();
     let mut field_get_components = Vec::new();
     let mut field_from_components = Vec::new();
-    for ((field_type, field_kind), field) in
-        field_type.iter().zip(field_kind.iter()).zip(field.iter())
+    for (((i, field_type), field_kind), field) in field_type
+        .iter()
+        .enumerate()
+        .zip(field_kind.iter())
+        .zip(field.iter())
     {
+        let index = syn::Index::from(i);
```

Comment:
Nit: Could this be moved inside the `None` match branch? Between ` None => {` and `field_get_components.push(quote! {`. So that it's defined closer to where it's used and easier to read. At first I thought "how could this compile without using `syn::Index::from`" but of course it's used here.